src/config/backend.ts

```diff
@@ -424,3 +439,223 @@ export class ExtlinuxConfigBackend extends DeviceConfigBackend {
+		// if we don't yet have a config file, just return an empty result...
+		if (!(await fs.exists(this.ConfigFilePath))) {
+			log.info('Empty ConfigFS config file');
+			return {};
+		}
+
+		// read the config file...
+		try {
+			const content = await fs.readFile(this.ConfigFilePath, 'utf8');
+			return JSON.parse(content);
+		} catch (err) {
+			log.error('Unable to deserialise ConfigFS configuration.', err);
+			return {};
+		}
+	}
+
+	private async writeConfigJSON(config: ConfigfsConfig): Promise<void> {
+		await remountAndWriteAtomic(this.ConfigFilePath, JSON.stringify(config));
+	}
+
+	private async loadConfiguredSsdt(config: ConfigfsConfig): Promise<void> {
+		if (config['ssdt'] !== undefined && _.isArray(config['ssdt'])) {
+			log.info('Loading configured SSDTs');
+			await Bluebird.mapSeries(config['ssdt'], aml => this.loadAML(aml));
+		}
+	}
+
+	public async initialise(
+		opts: BackendOptions,
+	): Promise<ConfigfsConfigBackend> {
+		try {
+			super.initialise(opts);
+
+			// load the acpi_configfs module...
+			await child_process.exec('modprobe acpi_configfs');
+
+			// read the existing config file...
+			const config = await this.readConfigJSON();
+
+			// write the config back out (reformatting it)
+			await this.writeConfigJSON(config);
+
+			// load the configured SSDT AMLs...
+			await this.loadConfiguredSsdt(config);
+			log.success('Initialised ConfigFS');
+		} catch (error) {
+			log.error(error);
+			if (this.options.logger) {
+				this.options.logger.logSystemMessage(
+					'Unable to initialise ConfigFS',
+					{ error },
+					'ConfigFS initialisation error',
+				);
+			}
+		}
+		return this;
+	}
+
+	public matches(deviceType: string): boolean {
+		return ConfigfsConfigBackend.SupportedDeviceTypes.includes(deviceType);
+	}
+
+	public async getBootConfig(): Promise<ConfigOptions> {
+		const options: ConfigOptions = {};
+
+		// read the config file...
+		const config = await this.readConfigJSON();
+
+		// see which SSDTs we have configured...
+		const ssdt = config['ssdt'];
+		if (_.isArray(ssdt) && ssdt.length > 0) {
+			// we have some...
+			options['ssdt'] = ssdt;
+		}
+		return options;
+	}
+
+	public async setBootConfig(opts: ConfigOptions): Promise<void> {
+		// read the config file...
+		const config = await this.readConfigJSON();
+
+		// see if the target state defined some SSDTs...
+		const ssdtKey = `${this.ConfigVarNamePrefix}ssdt`;
+		if (opts[ssdtKey]) {
+			// it did, so update the config with theses...
+			const ssdt = opts[ssdtKey];
+			const ssdts = _.isArray(ssdt) ? ssdt : [ssdt];
+			config['ssdt'] = ssdts;
+		} else {
+			// it did not, so remove any existing SSDTs from the config...
+			delete config['ssdt'];
+		}
+
+		// store the new config to disk...
+		await this.writeConfigJSON(config);
+	}
+
+	public isSupportedConfig(name: string): boolean {
+		switch (this.stripPrefix(name)) {
+			case 'ssdt':
+				return true;
+			default:
+				return false;
+		}
```

Comment:
This could also be
```suggestion
		return this.stripPrefix(name) === 'ssdt'
```
or if you really want to leave it potentially extendable then
```suggestion
		return ['ssdt'].includes(this.stripPrefix(name))
```
(I don't think it's necessary as it's an easy change to make when adding more anyway)

Reply:
agreed

Reply:
Btw @richbayliss the bit I felt was unnecessary was the `[].includes` part, I think moving away from the `switch` makes sense